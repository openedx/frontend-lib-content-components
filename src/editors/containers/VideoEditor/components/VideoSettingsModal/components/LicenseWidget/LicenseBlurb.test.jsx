import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { LicenseBlurb } from './LicenseBlurb';

describe('LicenseBlurb', () => {
  const props = {
    license: 'all-rights-reserved',
    details: {},
  };

  describe('snapshots', () => {
    test('snapshots: renders as expected with default props', () => {
      expect(
        shallow(<LicenseBlurb {...props} />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected with license equal to Creative Commons', () => {
      expect(
        shallow(<LicenseBlurb {...props} license="creative-commons" />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected when details.attribution equal true', () => {
      expect(
        shallow(<LicenseBlurb {...props} license="creative-commons" details={{ attribution: true }} />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected when details.attribution and details.noncommercial equal true', () => {
      expect(
        shallow(<LicenseBlurb {...props} license="creative-commons" details={{ attribution: true, noncommercial: true }} />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected when details.attribution and details.noDerivatives equal true', () => {
      expect(
        shallow(<LicenseBlurb {...props} license="creative-commons" details={{ attribution: true, noDerivatives: true }} />).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected when details.attribution and details.shareAlike equal true', () => {
      expect(
        shallow(<LicenseBlurb {...props} license="creative-commons" details={{ attribution: true, shareAlike: true }} />).snapshot,
      ).toMatchSnapshot();
    });
  });
});
