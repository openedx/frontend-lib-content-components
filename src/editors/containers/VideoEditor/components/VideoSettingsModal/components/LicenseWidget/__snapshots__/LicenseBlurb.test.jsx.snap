// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenseBlurb snapshots snapshots: renders as expected when details.attribution and details.noDerivatives equal true 1`] = `
<div
  className="d-flex flex-row align-items-center mt-2"
>
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <Icon
    className="mr-1 text-primary-300"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <div
    className="small mx-1.5"
  >
    <FormattedMessage
      defaultMessage="Some Rights Reserved"
      description="Label for row of creative common icons"
      id="authoring.videoeditor.license.creativeCommonsIcons.label"
    />
  </div>
</div>
`;

exports[`LicenseBlurb snapshots snapshots: renders as expected when details.attribution and details.noncommercial equal true 1`] = `
<div
  className="d-flex flex-row align-items-center mt-2"
>
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <Icon
    className="mr-1 text-primary-300"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <div
    className="small mx-1.5"
  >
    <FormattedMessage
      defaultMessage="Some Rights Reserved"
      description="Label for row of creative common icons"
      id="authoring.videoeditor.license.creativeCommonsIcons.label"
    />
  </div>
</div>
`;

exports[`LicenseBlurb snapshots snapshots: renders as expected when details.attribution and details.shareAlike equal true 1`] = `
<div
  className="d-flex flex-row align-items-center mt-2"
>
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <Icon
    className="mr-1 text-primary-300"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <div
    className="small mx-1.5"
  >
    <FormattedMessage
      defaultMessage="Some Rights Reserved"
      description="Label for row of creative common icons"
      id="authoring.videoeditor.license.creativeCommonsIcons.label"
    />
  </div>
</div>
`;

exports[`LicenseBlurb snapshots snapshots: renders as expected when details.attribution equal true 1`] = `
<div
  className="d-flex flex-row align-items-center mt-2"
>
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <Icon
    className="mr-1 text-primary-300"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <div
    className="small mx-1.5"
  >
    <FormattedMessage
      defaultMessage="Some Rights Reserved"
      description="Label for row of creative common icons"
      id="authoring.videoeditor.license.creativeCommonsIcons.label"
    />
  </div>
</div>
`;

exports[`LicenseBlurb snapshots snapshots: renders as expected with default props 1`] = `
<div
  className="d-flex flex-row align-items-center mt-2"
>
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <div
    className="small mx-1.5"
  >
    <FormattedMessage
      defaultMessage="All Rights Reserved"
      description="Label for row of all rights reserved icons"
      id="authoring.videoeditor.license.allRightsReservedIcons.label"
    />
  </div>
</div>
`;

exports[`LicenseBlurb snapshots snapshots: renders as expected with license equal to Creative Commons 1`] = `
<div
  className="d-flex flex-row align-items-center mt-2"
>
  <Icon
    className="mr-1"
    style={
      {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <div
    className="small mx-1.5"
  >
    <FormattedMessage
      defaultMessage="Some Rights Reserved"
      description="Label for row of creative common icons"
      id="authoring.videoeditor.license.creativeCommonsIcons.label"
    />
  </div>
</div>
`;
